### DISCUSSION TITLE
Autofill on Mull Browser

### DISCUSSION THREAD

**Discussion by euh2** [zeapo/Android-Password-Store#1638] [Q&A] [answered]
Mull is based on Firefox. But somehow Mull doesn't work properly with APS.

When i click a username textfield I would expect the APS pop-up to show 
```
"domain.com"
Search in store ...
```
Instead it says
```
"Mull"
Search in store ...
```
I guess APS doesn't get the current domain and therefore shows the app-name instead. 

Any idea how to fix that? Maybe some about:config setting?

**Comment by msfjarvis** [comment, marked as answer]
APS uses a hard-coded list of supported browsers for feature and trust detection as given [here](https://github.com/android-password-store/Android-Password-Store/blob/d6e89f04254ac77f17ca39c604234eb552d43952/autofill-parser/src/main/java/com/github/androidpasswordstore/autofillparser/FeatureAndTrustDetection.kt#L59-L90). If you can provide the link to a homepage and a source for release assets for Mull browser I can evaluate and include it.

**Reply by euh2** [reply]
Ok, I understand. Here is the link to the build scripts: https://github.com/Divested-Mobile/Mull-Fenix

As far as I know the scripts pull Firefox/ Fenix code from upstream, builds it and publishes on F-Droid.

**Reply by bratekarate** [reply]
Any updates? Seems like mull is still not in the list.

**Reply by bezmi** [reply]
> Seems like mull is still not in the list.

[It's included in a pre-release](https://github.com/android-password-store/Android-Password-Store/releases/tag/autofill-parser-v1.1.1) that you'll need to build yourself. It [will ~probably~ be](https://github.com/android-password-store/Android-Password-Store/blob/7f90d827ae436fe74ce5d3cabb9f1b091462b657/autofill-parser/src/main/java/com/github/androidpasswordstore/autofillparser/FeatureAndTrustDetection.kt#L91) included in the v2.0 release of the app.

**Reply by bratekarate** [reply]
Ok, fair enough. I thought the link in the accepted answer was to the master version of the list, but apparently it is just a specific revision. I will build the APK myself then, thanks for the hint.

**Reply by msfjarvis** [reply]
The linked pre-release is actually just for the Autofill code that is shipped as a distributable library for other Android apps. The plan is indeed for this to be included in v2.0 of APS, which will **not** be shipped as an upgrade and instead will be a different app altogether.

Re: building the app yourself, 1) there's really no need, there are [snapshot builds available](https://docs.passwordstore.app/docs/users/release-channels); and 2) the development branch uses an entirely different, work-in-progress PGP backend that has a fairly poor onboarding flow at the moment and thus you may find it slightly confusing to set things up with it.
